git-extra/git-askyesno.c

```diff
@@ -0,0 +1,23 @@
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+int wmain(int argc, wchar_t **wargv)
+{
+	int res;
+	wchar_t *title = L"Question?";
+
+	if (argc > 2 && !wcscmp(L"--title", wargv[1])) {
+		title = wargv[2];
+		argc -=2;
+		wargv += 2;
+	}
+
+	if (argc != 2) {
+		MessageBoxW(NULL, L"Required: exactly one argument", L"Error!", MB_OK);
+		return 1;
+	}
+
```

Comment:
You've dropped the part where the TCL script concatenated all arguments (after the title). Shouldn't matter too much.

Reply:
I'd have done something like this:

````C
char *prompt=NULL;
//...
for(int j = i; j < argc; j++)
    len += strlen(argv[j])+1;
prompt = (char*)malloc(len);
if(prompt != NULL)
{
      strcpy(prompt,argv[i]);
      for(int j = i+1; j < argc; j++)
      {
           strcat(prompt," ");
           strcat(prompt, argv[j]);
      }
//messagepox here
free(prompt);
}
else
    return 1;
````

though slightly modified for wchar_t instead of char.

Reply:
Good point. Let me copy/butcher this.

Reply:
Butchered and force-pushed. Could you have a look?